sim/src/node.rs

```diff
@@ -76,12 +76,15 @@ impl SimElement for Node {
                 for n in &self.neighbors {
                     if n == &dest {
                         have_dest = true;
-                        ret.push((rpc, dest));
+                        ret.push((rpc, dest.clone()));
                         break;
                     }
                 }
                 if !have_dest {
-                    print!("WARNING:  RPC given with invalid destination\n");
+                    print!(
+                        "WARNING:  RPC given with invalid destination {0}\n",
+                        dest.clone()
```

Comment:
Cloning for a print seems wrong.

Reply:
We have to either clone here or at ret.push as was below.  This is because we are using dest both for the print and moving it into ret.  I think it is better that we clone in the actual code than for a print, so I reverted the below change and got rid of the clone here.